src/expression.js

```diff
@@ -565,6 +567,30 @@ pp.parseObj = function(isPattern, refDestructuringErrors) {
 
 pp.parseProperty = function(isPattern, refDestructuringErrors) {
   let prop = this.startNode(), isGenerator, isAsync, startPos, startLoc
+  if (this.options.ecmaVersion >= 9 && this.eat(tt.ellipsis)) {
+    // To disallow parenthesized identifier via `this.toAssignable()`.
+    if (this.type === tt.parenL && refDestructuringErrors) {
```

Comment:
👆 just to clarify the check above
> To disallow parenthesized identifier via `this.toAssignable()`

Is that the default param value guard? Or something else *(does it have a test)*.

Reply:
This is the guard for the parenthesized identifier in RestElement. E.g. `({...(obj)} = foo)`. Yes, there are tests.